network/gateway/base.go

```diff
@@ -61,27 +61,34 @@ import (
 // Base is abstract class for gateways
 
 type Base struct {
-	Network            network.Gatewayer
-	GIL                insolar.GlobalInsolarLock
-	SwitcherWorkAround insolar.SwitcherWorkAround // nodekeeper
+	Me                  network.Gateway
```

Comment:
`This` or `Self` are more appropriate imho

Reply:
gut!